core/arch/arm/include/kernel/spinlock.h

```diff
@@ -81,6 +81,21 @@ static inline void cpu_spin_unlock(unsigned int *lock)
 	__cpu_spin_unlock(lock);
 	spinlock_count_decr();
 }
+
+static inline uint32_t cpu_spin_lock_xsave(unsigned int *lock, uint32_t mask)
```

Comment:
I think the `mask` flag isn't needed as mask always is `THREAD_EXCP_ALL`. If we'd ever have an odd case where it should be different we'll deal with that then.

Update: I see that we're not using `THREAD_EXCP_ALL` in malloc below. It should either be changed or explained why not with a comment because it's at least not obvious to me.